x/nft/types/codec.go

```diff
@@ -8,10 +8,10 @@ var cdc = codec.New()
 
 // RegisterCodec concrete types on codec
 func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterInterface((*NFT)(nil), nil)
+	cdc.RegisterConcrete(&BaseNFT{}, "cosmos-sdk/BaseNFT", nil)
 	cdc.RegisterConcrete(MsgTransferNFT{}, "cosmos-sdk/MsgTransferNFT", nil)
 	cdc.RegisterConcrete(MsgEditNFTMetadata{}, "cosmos-sdk/MsgEditNFTMetadata", nil)
-	cdc.RegisterConcrete(MsgMintNFT{}, "cosmos-sdk/MsgMintNFT", nil)
-	cdc.RegisterConcrete(MsgBurnNFT{}, "cosmos-sdk/MsgBurnNFT", nil)
```

Comment:
@fedekunze If we leave the msg types in the module (although we don't allow them to do anything until called from within an app) should we still register them?

Reply:
no, we just register the types and interfaces that we use. The rest is up to the user. We should still allow to register new types into the NFT codec tho, I'll add that to the TODOs